src/pages/outwards-return/index.js

```diff
@@ -0,0 +1,162 @@
+import { getStorageData } from 'src/storage/storage'
+import { RemittanceOutwardsRepository } from 'src/repositories/RemittanceOutwardsRepository'
+import OutwardsReturnForm from './Forms/OutwardsReturnForm'
+import GridToolbar from 'src/components/Shared/GridToolbar'
+import { SystemFunction } from 'src/resources/SystemFunction'
+import { useDocumentTypeProxy } from 'src/hooks/documentReferenceBehaviors'
+
+const OutwardsReturn = () => {
+  const { getRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+  const { stack: stackError } = useError()
+  const { stack } = useWindow()
+
+  async function fetchGridData(options = {}) {
+    const { _startAt = 0, _pageSize = 50, params } = options
+
+    const response = await getRequest({
+      extension: RemittanceOutwardsRepository.OutwardsReturn.qry,
+      parameters: `_startAt=${_startAt}&_pageSize=${_pageSize}&_params=${params || ''}&filter=`
+    })
+
+    return { ...response, _startAt: _startAt }
+  }
+
+  const {
+    query: { data },
+    labels: _labels,
+    paginationParameters,
+    refetch,
+    access
+  } = useResourceQuery({
+    queryFn: fetchGridData,
+    endpointId: RemittanceOutwardsRepository.OutwardsReturn.qry,
+    datasetId: ResourceIds.OutwardsReturn,
+  })
+
+  const columns = [
+    {
+      field: 'reference',
+      headerName: _labels.reference,
+      flex: 1
+    },
+    {
+      field: 'date',
+      headerName: _labels.date,
+      flex: 1,
+      type: 'date'
+    },
+    {
+      field: 'outwardRef',
+      headerName: _labels.outwardRef,
+      flex: 1
+    },
+    {
+      field: 'requestedByName',
+      headerName: _labels.requestedBy,
+      flex: 1
+    }
+  ]
+
+  const { proxyAction } = useDocumentTypeProxy({
+    functionId: SystemFunction.OutwardsReturn,
+    action: openForm,
+    hasDT: false
+  })
+
+  const add = async () => {
+    await proxyAction()
+  }
+
+  const edit = obj => {
+    openForm(obj?.recordId)
+  }
+
+  const getPlantId = async () => {
+    const userId = getStorageData('userData').userId
+
+    try {
+      const res = await getRequest({
+        extension: SystemRepository.UserDefaults.get,
+        parameters: `_userId=${userId}&_key=plantId`
+      })
+
+      return res.record.value
+    } catch (e) {}
+  }
+
+  async function openOutWardsWindow(plantId, recordId) {
+    const dtId = await getDefaultDT()
+
+    stack({
+      Component: OutwardsReturnForm,
+      props: {
+        labels: _labels,
+        recordId,
+        plantId,
+        maxAccess: access,
+        dtId
+      },
+      width: 600,
+      height: 600,
+      title: _labels.outwardsReturn
+    })
+  }
+
+  async function openForm(recordId) {
+    const plantId = await getPlantId()
+
+    plantId !== ''
+      ? openOutWardsWindow(plantId, recordId)
+      : stackError({
+          message: platformLabels.noDefaultPlant
+        })
+  }
+
+  const userData = getStorageData('userData')
```

Comment:
replace this by `const userId = getStorageData('userData').userId`
and use it in both getPlantId & getDefaultDT to prevent duplicate (since in getPlantId we also add const for userId)